tests/system/large/test_location.py

```diff
@@ -0,0 +1,184 @@
+    "europe-west1",
+    "europe-west10",
+    "europe-north1",
+    "europe-west3",
+    "europe-west2",
+    "europe-southwest1",
+    "europe-west8",
+    "europe-west4",
+    "europe-west9",
+    "europe-west12",
+    "europe-central2",
+    "europe-west6",
+    "me-central2",
+    "me-central1",
+    "me-west1",
+    "me-central2",
+    "me-central1",
+    "me-west1",
+    "africa-south1",
+]
+
+REP_ENABLED_BIGQUERY_LOCATIONS = [
+    "me-central2",
+    "europe-west9",
+    "europe-west3",
+    "us-east4",
+    "us-west1",
+]
+
+
+def _assert_bq_execution_location(session: bigframes.Session):
+    df = session.read_gbq(
+        """
+        SELECT "aaa" as name, 111 as number
+        UNION ALL
+        SELECT "bbb" as name, 222 as number
+        UNION ALL
+        SELECT "aaa" as name, 333 as number
+    """
+    )
+
+    assert (
+        typing.cast(bigquery.QueryJob, df.query_job).location
+        == session.bqclient.location
+    )
+
+    result = (
+        df[["name", "number"]]
+        .groupby("name")
+        .sum(numeric_only=True)
+        .sort_values("number", ascending=False)
+        .head()
+    )
+
+    assert (
+        typing.cast(bigquery.QueryJob, result.query_job).location
+        == session.bqclient.location
+    )
+
+
+def test_bq_location_default():
+    session = bigframes.Session()
+
+    assert session.bqclient.location == "US"
+
+    # by default global endpoint is used
+    assert (
+        session.bqclient._connection.API_BASE_URL == "https://bigquery.googleapis.com"
+    )
+
+    # assert that bigframes session honors the location
+    _assert_bq_execution_location(session)
+
+
+@pytest.mark.parametrize("bigquery_location", ALL_BIGQUERY_LOCATIONS)
+def test_bq_location(bigquery_location):
+    session = bigframes.Session(
+        context=bigframes.BigQueryOptions(location=bigquery_location)
+    )
+
+    assert session.bqclient.location == bigquery_location
+
+    # by default global endpoint is used
+    assert (
+        session.bqclient._connection.API_BASE_URL == "https://bigquery.googleapis.com"
+    )
+
+    # assert that bigframes session honors the location
+    _assert_bq_execution_location(session)
+
+
+@pytest.mark.parametrize(
+    "bigquery_location",
+    REP_ENABLED_BIGQUERY_LOCATIONS,
+)
+def test_bq_rep_endpoints(bigquery_location):
+    session = bigframes.Session(
+        context=bigframes.BigQueryOptions(
+            location=bigquery_location, use_regional_endpoints=True
+        )
+    )
+
+    assert session.bqclient.location == bigquery_location
+    assert (
+        session.bqclient._connection.API_BASE_URL
+        == "https://bigquery.{location}.rep.googleapis.com".format(
+            location=bigquery_location
+        )
+    )
+
+    # assert that bigframes session honors the location
+    _assert_bq_execution_location(session)
+
+
+@pytest.mark.parametrize(
+    "bigquery_location",
+    set(ALL_BIGQUERY_LOCATIONS) - set(REP_ENABLED_BIGQUERY_LOCATIONS),
```

Comment:
Might need to add `sorted` here to avoid the error.

```suggestion
    sorted(set(ALL_BIGQUERY_LOCATIONS) - set(REP_ENABLED_BIGQUERY_LOCATIONS)),
```